src/lmcontrol/nn/clf.py

```diff
@@ -0,0 +1,484 @@
+import argparse
+import glob
+import pickle
+import os
+
+from functools import partial
+import lightning as L
+import torch
+from torch.utils.data import DataLoader
+import torchvision
+from torch import nn
+import torch
+import glob
+import numpy as np
+
+import optuna
+import lightning as L
+
+import wandb
+from lightning.pytorch.loggers import WandbLogger, CSVLogger
+from sklearn.metrics import accuracy_score, precision_score, recall_score, confusion_matrix
+from lightning.pytorch.callbacks import EarlyStopping
+from sklearn.preprocessing import LabelEncoder
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+from lightning.pytorch.callbacks import ModelCheckpoint
+from torchvision.models.resnet import BasicBlock, Bottleneck 
+from typing import Type, Union, List, Optional, Callable, Any
+from optuna.integration import PyTorchLightningPruningCallback
+
+from ..utils import get_logger
+from ..data_utils import load_npzs, encode_labels
+from .dataset import LMDataset, get_transforms as _get_transforms
+from lmcontrol.nn.resnet import _resnet  
+
+def resnet(*, weights=None, progress=True, block=None, layers=None, planes=None,num_classes=None, save_embeddings=None) :
+    """ResNet-18 from `Deep Residual Learning for Image Recognition <https://arxiv.org/abs/1512.03385>`__.
+
+    Args:
+        weights (:class:`~torchvision.models.ResNet18_Weights`, optional): The
+            pretrained weights to use. See
+            :class:`~torchvision.models.ResNet18_Weights` below for
+            more details, and possible values. By default, no pre-trained
+            weights are used.
+        progress (bool, optional): If True, displays a progress bar of the
+            download to stderr. Default is True.
+        **kwargs: parameters passed to the ``torchvision.models.resnet.ResNet``
+            base class. Please refer to the `source code
+            <https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py>`_
+            for more details about this class.
+
+    .. autoclass:: torchvision.models.ResNet18_Weights
+        :members:
+    """
+    return _resnet(block=block, layers=layers, planes=planes, num_classes=num_classes, weights=weights, progress=progress, save_embeddings=save_embeddings)
+
+
+class LightningResNet(L.LightningModule):
+
+    val_metric = "validation_ncs"
+    train_metric = "train_ncs"
+
+    def __init__(self, num_classes, lr=0.01, step_size=2, gamma=0.1, planes=[8, 16, 32, 64], layers=[1, 1, 1, 1], block=BasicBlock, save_embeddings=False):
+        super().__init__()
+
+        weights = None 
+        progress = True
+
+        self.backbone = resnet(weights=None, progress=True, block=block, layers=layers, planes=planes,num_classes=num_classes, save_embeddings=save_embeddings) 
+        self.criterion = nn.MSELoss()
```

Comment:
Why is this MSELoss?

Reply:
I am computing the MSE loss just for the purpose of comparison of behaviour of model on the different datasets. I am not storing the losses of each image.